### PR TITLE
Fix formatting of dicts

### LINKED ISSUE #12: Dictionaries not converted properly
Hi @asottile 

Great package, using it to upgrade all my projects to declarative metadata right now.

I've had one issue with dictionaries beingn converted to their `repr()`:

```
$ cat setup.py
from setuptools import setup

setup(project_urls={'homepage': 'https://adamj.eu'})
$ setup-py-upgrade .
./setup.py and ./setup.cfg written!
$ cat setup.py
from setuptools import setup
setup()
$ cat setup.cfg
[metadata]
project_urls = {'homepage': 'https://adamj.eu'}
```

The expected output for dictionaries is `key = value`, one per line, like:

```
[metadata]
project_urls =
    homepage = https://adamj.eu
```

It's not a bother for me, I can fix my projects by hand. But it does lead to a slightly confusing error which users might need a little time to debug from the message and the setup.cfg docs.

Thanks,

Adam

### PR DESCRIPTION
**Pull request by adamchainz**
Fixes #12. Tested using the example I gave there.

### CODE REVIEW

File: setup_py_upgrade.py [hunk 1]
```diff
@@ -134,11 +134,23 @@ def _list_as_str(lst: Sequence[str]) -> str:
...
 
 
-def reformat_lists(section: Dict[str, Any]) -> Dict[str, Any]:
-    return {
-        k: _list_as_str(v) if isinstance(v, (list, tuple)) else v
-        for k, v in section.items()
-    }
+def _dict_as_str(dct: Dict[str, str]) -> str:
+    return '\n' + '\n'.join(
+        (key + ' = ' + value) for key, value in dct.items()
```
Comment (asottile) [inline setup_py_upgrade.py]: `f'{key} = {value}'`

File: setup_py_upgrade.py [hunk 2]
```diff
@@ -134,11 +134,23 @@ def _list_as_str(lst: Sequence[str]) -> str:
...
+    )
+
+
+def reformat_values(section: Dict[str, Any]) -> Dict[str, Any]:
```
Comment (asottile) [inline setup_py_upgrade.py]: maybe name this `_reformat` -- now that it's doing a bunch of things the original name made no sense and I don't think `_values` adds any value ;)

File: setup_py_upgrade.py [hunk 3]
```diff
@@ -134,11 +134,23 @@ def _list_as_str(lst: Sequence[str]) -> str:
...
+    new_section = {}
+    for key, value in section.items():
+        if isinstance(value, (list, tuple)):
+            new_value = _list_as_str(value)
+        elif isinstance(value, dict):
+            new_value = _dict_as_str(value)
+        else:
+            new_value = value
```
Comment (asottile) [inline setup_py_upgrade.py]: it'd be slightly shorter to just `new_section[key] = ...` for each of the three branches (less state to think about as well)

**Comment by asottile** [comment]
(not your fault, I didn't have a testsuite set up here yet -- but this broke `package_dir` handling -- I'm adding a testsuite in #14)

**Comment by asottile** [comment]
(I originally wrote this [on stream](https://www.youtube.com/watch?v=vGyshCvMB-M&t=6980s) and didn't know whether it would work so tests were a bit of an afterthought)

**Comment by adamchainz** [comment]
Oh sorry about that.

Fun that you streamed it!

**Comment by asottile** [comment]
no problem! thanks for the patch \o/

**Pull request merged**